### PR TITLE
Kubernetes deploy

### PR DESCRIPTION
**Pull request by R00tedSec**
# Kubernetes deploy

Using this manifests trudesk can be deployed on a kubernetes cluster

### CODE REVIEW

**Comment by polonel** [comment]
I don't have my config in front of me cause I'm at work, but you need a **config-map** set for `config.json` which looks something like:
```yaml
apiVersion: v1
data:
  config.json: |-
    {
      "tokens": {
        "secret": "SECRET_USED_TO_HASH"
      }
    }
kind: ConfigMap
metadata:
  name: trudesk-config
  namespace: default
```
Volume mount the config-map to `/usr/src/trudesk/config.json` and the subpath/key in the config-map `config.json` and mark as read-only.

This will also fix the issue you're having with trudesk running the install/setup after the pod dies.

File: kubernetes/mongo-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            - mountPath: /data/db
+              name: mongo-claim0
+      restartPolicy: Always
+      volumes:
+        - name: mongo
+          persistentVolumeClaim:
+            claimName: mongo
+        - name: mongo-claim0
+          persistentVolumeClaim:
+            claimName: mongo-claim0
```
Comment (R00tedSec) [inline kubernetes/mongo-deployment.yaml:59]: Mongodb was wiping the data every time it was recreated, forcing trudesk to restart de installation process

Comment (R00tedSec) [inline kubernetes/mongo-deployment.yaml:59]: Mongodb was wiping the data every time it was recreated, forcing trudesk to restart de installation process

**Comment by R00tedSec** [comment]
Mongodb was wiping the data every time it was recreated, forcing trudesk to restart the installation process

Adding config.json and this code to the mongo-deployment.yaml :
```yaml      
    volumeMounts:
            - mountPath: /var/lib/mongo
              name: mongo
            - mountPath: /data/db
              name: mongo-claim0
      restartPolicy: Always
      volumes:
        - name: mongo
          persistentVolumeClaim:
            claimName: mongo
        - name: mongo-claim0
          persistentVolumeClaim:
            claimName: mongo-claim0
```
Trudesk works 100% fine.

**Pull request merged**